### DISCUSSION TITLE
homebrew Python version &  `uv tool install --upgrade mflux` issue -> documentation note?

### DISCUSSION THREAD

**Discussion by gmorain** [filipstrand/mflux#541] [Ideas & Suggestions: Features]
Hi,

I had an issue updating MFlux to 0.5.0 this morning, as the `tool install --upgrade mflux` command failed. 

Homebrew is quite 'aggressive' in keeping all package versions up to date, so after a recent `brew install <something>` my setup was pushed to Python 3.13 (as seen with the `python3 --version` and `uv python list` commands).

Having met this kind of breaking behaviour before, and having seen the Python 3.13 note and issue in the MFlux README / repo, I reverted my default homebrew Python version to 3.9 and voilà! (uv still uses the recommended Python version defined in the toml file coming with mflux (ie something >=3.10 if I am not mistaken).

```bash
brew unlink python@3.9
brew unlink python@3.13
brew link --force --overwrite python@3.9
uv tool install --upgrade mflux
```

Hope this helps people getting the same issue while using homebrew and MFlux

**Comment by filipstrand** [comment]
Thanks for sharing your issue and strategy here. I have not personally kept up with the state of python 3.13 yet, so cannot comment about that, but happy it worked by downgrading the python version!

I am happy to keep this issue open as a reference for others and I'll just make a small general note in the readme to point people to the issue section if they have any such troubles (a while back, I was thinking about adding a general "installation issues" sub section in the README, but it feels like there are so many ways things can break and differ between systems and setups, so I'll keep that section very brief and simply point people to the issue section)

**Comment by gmorain** [comment]
Hi @filipstrand, having so many possible causes for trouble in this fast-moving space might indeed make the README grow beyond a manageable size, so I totally understand your point in keeping that in a separate place. May I suggest the introduction of tags so issues can be listed as "documentation", "improvement", and so on?

**Comment by filipstrand** [comment]
That is a good idea

**Comment by wschenk** [comment]
This is super easy.

```
uv tool install -p 3.12 --upgrade mflux
```
